whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/importer/DeltaImporter.java

```diff
@@ -0,0 +1,175 @@
+package net.ripe.db.nrtm4.client.importer;
+
+import io.netty.util.internal.StringUtil;
+import net.ripe.db.nrtm4.client.client.MirrorDeltaInfo;
+import net.ripe.db.nrtm4.client.client.NrtmRestClient;
+import net.ripe.db.nrtm4.client.client.UpdateNotificationFileResponse;
+import net.ripe.db.nrtm4.client.condition.Nrtm4ClientCondition;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientInfoRepository;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientRepository;
+import net.ripe.db.nrtm4.client.dao.NrtmClientVersionInfo;
+import net.ripe.db.whois.common.dao.RpslObjectUpdateInfo;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.apache.commons.compress.utils.Lists;
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+@Service
+@Conditional(Nrtm4ClientCondition.class)
+public class DeltaImporter implements Importer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeltaImporter.class);
+
+    private final NrtmRestClient nrtmRestClient;
+
+    private final Nrtm4ClientRepository nrtm4ClientRepository;
+
+    private final Nrtm4ClientInfoRepository nrtm4ClientInfoRepository;
+
+
+    public static final String RECORD_SEPARATOR = "\u001E";
+
+    public DeltaImporter(final NrtmRestClient nrtmRestClient,
+                         final Nrtm4ClientRepository nrtm4ClientRepository,
+                         final Nrtm4ClientInfoRepository nrtm4ClientInfoRepository) {
+        this.nrtmRestClient = nrtmRestClient;
+        this.nrtm4ClientRepository = nrtm4ClientRepository;
+        this.nrtm4ClientInfoRepository = nrtm4ClientInfoRepository;
+    }
+
+    @Override
+    public void doImport(final String source, final UpdateNotificationFileResponse updateNotificationFile){
+        final List<UpdateNotificationFileResponse.NrtmFileLink> deltas = getNewDeltas(source, updateNotificationFile);
+
+        if (deltas.isEmpty()) {
+            LOGGER.info("No new deltas to be processed");
+            return;
+        }
+
+        deltas.forEach(delta -> {
+            final byte[] deltaFilePayload = nrtmRestClient.getDeltaFiles(delta.getUrl());
+
+            if (deltaFilePayload == null || deltaFilePayload.length == 0){
+                LOGGER.error("This cannot happen. UNF has a non-existing delta");
+                return;
+            }
+
+            final String[] deltaFileResponse = StringUtils.split(new String(deltaFilePayload, StandardCharsets.UTF_8), RECORD_SEPARATOR);
```

Comment:
You're using a different library to split the String than the rest of the application which uses Guava Splitter. Also you have a `byte[]` array which you turn into a big string then immediately throw it away. You could iterate through the `byte[]` and split on the byte value 0x1E directly

Reply:
e.g. use `ByteBuffer.wrap()` then interate and use `ByteBuffer.slice()` to generate Strings. Or maybe there's a library method to slice arrays for you.

Reply:
yes, I am doing that in snapshot already. For deltas I though that it was not really necessary because usually there are not too much deltas. 
I will refactor this

Reply:
Done. I am processing each line of the delta file without transforming everything into string